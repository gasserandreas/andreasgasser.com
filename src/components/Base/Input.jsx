import React from 'react';

import { Input as RInput } from '@rebass/forms/styled-components';
import styled from 'styled-components';

const Styles = {
  Input: styled(RInput)(({ theme }) => `
    background-color: none;
    border: none;
    border-bottom: 1.25px solid ${theme.colors.muted};
    padding: 0.25rem;
    font-size: 0.95rem;
    line-height: 1.75rem;
  `),
};

const Input = (props) => <Styles.Input {...props} />;

export default Input;
