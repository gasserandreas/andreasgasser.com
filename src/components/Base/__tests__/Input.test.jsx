/* global withTheme */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Input from '../Input';

describe('components/Input', () => {
  it('should render', () => {
    const dom = mount(withTheme(<Input />));

    expect(dom.exists()).toBeTruthy();

    expect(toJson(dom)).toMatchSnapshot();
  });
});
