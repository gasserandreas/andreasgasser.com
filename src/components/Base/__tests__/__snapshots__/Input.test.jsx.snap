// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Input should render 1`] = `
<CustomThemeProvider
  dark={false}
>
  <ThemeProvider
    theme={
      Object {
        "breakpoints": Array [
          "320px",
          "540px",
          "720px",
          "960px",
          "1140px",
          "1400px",
        ],
        "buttons": Object {
          "link": Object {
            "&:hover": Object {
              "color": "muted",
              "cursor": "pointer",
            },
            "backgroundColor": "inherit",
            "border": "none",
            "color": "text",
            "fontSize": "0.95rem",
            "fontWeight": 400,
            "height": "2.5rem",
            "padding": "0rem 0.5rem",
            "textTransform": "uppercase",
          },
          "outline": Object {
            "&:hover": Object {
              "backgroundColor": "text",
              "color": "textInverse",
              "cursor": "pointer",
            },
            "backgroundColor": "inherit",
            "border": "1px solid #000",
            "borderColor": "text",
            "color": "text",
            "fontSize": "0.95rem",
            "fontWeight": 400,
            "height": "2.5rem",
            "padding": "0rem 0.5rem",
            "textTransform": "uppercase",
          },
        },
        "colors": Object {
          "background": "#c3c3c3",
          "backgroundInverse": "#fff",
          "error": "#E11531",
          "muted": "#595959",
          "text": "#333333",
          "textHighlighed": "#333",
          "textHover": "#7e7e7e",
          "textInverse": "#f6f6f6",
        },
        "fontSizes": Array [
          14,
          16,
          20,
          24,
          28,
          32,
          40,
          48,
          56,
          64,
        ],
        "fontWeights": Object {
          "body": 400,
          "bold": 900,
          "heading": 600,
        },
        "fonts": Object {
          "body": "Open Sans, Helvetica Neue, Helvetica, sans-serif",
          "heading": "inherit",
          "monospace": "Menlo, monospace",
        },
        "lineHeights": Object {
          "body": 1.5,
          "heading": 1.125,
        },
        "radii": Object {
          "default": "2px",
        },
        "shadows": Object {
          "card": "0 0 4px rgba(0, 0, 0, .125)",
        },
        "sizes": Object {
          "avatar": 48,
        },
        "space": Array [
          0,
          4,
          8,
          16,
          32,
          64,
          128,
          256,
          512,
        ],
        "styles": Object {
          "root": Object {
            "fontFamily": "body",
            "fontWeight": "body",
            "lineHeight": "body",
          },
        },
        "text": Object {
          "bold": Object {
            "color": "#333333",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
            "fontWeight": "900",
          },
          "default": Object {
            "color": "#333333",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
          },
          "muted": Object {
            "color": "#595959",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
          },
          "page": Object {
            "color": "#333333",
            "fontSize": Array [
              4,
              4,
              4,
              5,
              5,
              5,
            ],
            "fontWeight": 600,
            "letterSpacing": "0.075rem",
            "lineHeight": "2.75rem",
            "margin": "2rem 0 2.5rem",
            "maxWidth": "32rem",
            "wordSpacing": "-0.08em",
          },
          "section": Object {
            "color": "#f6f6f6",
            "fontSize": Array [
              1,
              1,
              1,
              2,
              2,
              2,
            ],
            "fontWeight": 900,
            "letterSpacing": "2px",
            "lineHeight": "1.7rem",
            "margin": "1.5rem 0",
            "textTransform": "uppercase",
          },
          "small": Object {
            "color": "#333333",
            "fontSize": Array [
              0,
              0,
              0,
              1,
              1,
              1,
            ],
          },
          "subPage": Object {
            "color": "#595959",
            "fontSize": Array [
              2,
              2,
              2,
              3,
              3,
              3,
            ],
            "fontWeight": 500,
            "letterSpacing": "0.0625rem",
            "lineHeight": "2.2rem",
            "margin": "1.25rem 0 1.75rem",
            "marginBottom": 2,
            "maxWidth": "32rem",
          },
        },
        "variants": Object {
          "avatar": Object {
            "borderRadius": "circle",
            "height": "avatar",
            "width": "avatar",
          },
          "card": Object {
            "bg": "background",
            "boxShadow": "card",
            "p": 2,
          },
          "link": Object {
            "color": "primary",
          },
          "nav": Object {
            ":hover,:focus,.active": Object {
              "color": "primary",
            },
            "color": "inherit",
            "display": "inline-block",
            "fontSize": 1,
            "fontWeight": "bold",
            "p": 2,
            "textDecoration": "none",
          },
        },
      }
    }
  >
    <Input>
      <Styled(Component)>
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": false,
                "lastClassName": "kNYspy",
                "rules": Array [
                  [Function],
                ],
              },
              "displayName": "Styled(Component)",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "render": [Function],
              },
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
        >
          <ForwardRef
            className="sc-htpNat kNYspy"
          >
            <styled.div
              __css={
                Object {
                  "appearance": "none",
                  "bg": "transparent",
                  "border": "1px solid",
                  "borderRadius": "default",
                  "color": "inherit",
                  "display": "block",
                  "fontSize": "inherit",
                  "lineHeight": "inherit",
                  "p": 2,
                  "width": "100%",
                }
              }
              as="input"
              className="sc-htpNat kNYspy"
              tx="forms"
              type="text"
              variant="input"
            >
              <StyledComponent
                __css={
                  Object {
                    "appearance": "none",
                    "bg": "transparent",
                    "border": "1px solid",
                    "borderRadius": "default",
                    "color": "inherit",
                    "display": "block",
                    "fontSize": "inherit",
                    "lineHeight": "inherit",
                    "p": 2,
                    "width": "100%",
                  }
                }
                as="input"
                className="sc-htpNat kNYspy"
                forwardedComponent={
                  Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "attrs": Array [],
                    "componentStyle": ComponentStyle {
                      "componentId": "sc-bdVaJa",
                      "isStatic": false,
                      "lastClassName": "gEBDmG",
                      "rules": Array [
                        "box-sizing: border-box;",
                        "margin: 0;",
                        "min-width: 0;",
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                        [Function],
                      ],
                    },
                    "displayName": "styled.div",
                    "foldedComponentIds": Array [],
                    "render": [Function],
                    "styledComponentId": "sc-bdVaJa",
                    "target": "div",
                    "toString": [Function],
                    "warnTooManyClasses": [Function],
                    "withComponent": [Function],
                  }
                }
                forwardedRef={null}
                tx="forms"
                type="text"
                variant="input"
              >
                <input
                  className="sc-bdVaJa gEBDmG sc-htpNat kNYspy"
                  type="text"
                />
              </StyledComponent>
            </styled.div>
          </ForwardRef>
        </StyledComponent>
      </Styled(Component)>
    </Input>
  </ThemeProvider>
</CustomThemeProvider>
`;
